vector/src/main/java/im/vector/home/NewHomeFragment.kt

```diff
@@ -0,0 +1,185 @@
+        val isFavorite = tags != null && tags.contains(RoomTag.ROOM_TAG_FAVOURITE)
+        val isLowPriority = tags != null && tags.contains(RoomTag.ROOM_TAG_LOW_PRIORITY)
+        RoomUtils.displayPopupMenu(activity, mSession, room, v, isFavorite, isLowPriority, this)
+    }
+
+    override fun onSelectRoom(room: Room?, position: Int) {
+        openRoom(room)
+    }
+
+    override fun onRegister(listener: UpDateListener) {
+        dataUpdateListeners.add(listener)
+    }
+
+    override fun onUnregister(listener: UpDateListener) {
+        dataUpdateListeners.remove(listener)
+    }
+
+    inner class HomePagerAdapter(fm: FragmentManager, val titles: Array<String>) : FragmentPagerAdapter(fm) {
+        private val pinMissedNotifications = PreferencesManager.pinMissedNotifications(activity)
+        private val pinUnreadMessages = PreferencesManager.pinUnreadMessages(activity)
+        private val notificationComparator = RoomUtils.getNotifCountRoomsComparator(mSession, pinMissedNotifications, pinUnreadMessages)
+
+        override fun getItem(position: Int): Fragment {
+            val fragment = when (position) {
+                ROOM_FRAGMENTS.INVITE.ordinal -> {
+                    val fragment = InviteRoomFragment()
+                    fragment.onUpdate(mActivity.roomInvitations, notificationComparator)
+                    fragment.onSelectRoomListener = this@NewHomeFragment
+                    fragment.invitationListener = this@NewHomeFragment
+                    fragment.moreActionListener = null
+                    fragment
+                }
+                ROOM_FRAGMENTS.FAVORITE.ordinal -> {
+                    val fragment = FavoriteRoomFragment()
+                    fragment.onUpdate(result?.favourites, notificationComparator)
+                    fragment.onSelectRoomListener = this@NewHomeFragment
+                    fragment.invitationListener = null
+                    fragment.moreActionListener = this@NewHomeFragment
+                    fragment
+                }
+                ROOM_FRAGMENTS.NORMAL.ordinal -> {
+                    val fragment = NormalRoomFragment()
+                    fragment.onUpdate(result?.otherRooms, notificationComparator)
+                    fragment.onSelectRoomListener = this@NewHomeFragment
+                    fragment.invitationListener = null
+                    fragment.moreActionListener = this@NewHomeFragment
+                    fragment
+                }
+                ROOM_FRAGMENTS.LOW_PRIORITY.ordinal -> {
+                    val fragment = LowPriorityRoomFragment()
+                    fragment.onUpdate(result?.lowPriorities, notificationComparator)
+                    fragment.onSelectRoomListener = this@NewHomeFragment
+                    fragment.invitationListener = null
+                    fragment.moreActionListener = this@NewHomeFragment
+                    fragment
+                }
+                else -> InviteRoomFragment()
```

Comment:
If there is no need to do the similar processing as above in lines 134 to 139, please explain why in a code comment

Reply:
I have added comment

Reply:
I do not think it's possible because we can not change the definition of the function.

Reply:
Sorry I meant throw an exception (not return an exception)

Reply:
I actually didn't realize I could do that. its done.